### PR TITLE
`solvers`: Implement CLI for 1inch

### PR DESCRIPTION
**Pull request by cowfee**
Progress on https://github.com/cowprotocol/services/issues/882.

Add a CLI command to run the 1inch solver. More or less just boilerplate code.

Follow-up to https://github.com/cowprotocol/services/pull/1250.

### Test Plan

Automated tests.

### CODE REVIEW

File: crates/solvers/src/infra/config/dex/oneinch/file.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    #[serde_as(as = "serialize::ChainId")]
+    chain_id: eth::ChainId,
+
+    /// The URL endpoint for the 1inch API.
+    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
+    endpoint: Option<reqwest::Url>,
+
+    /// The 1Inch liquidity sources to consider when swapping.
+    #[serde(flatten)]
+    liquidity: Liquidity,
```
Comment (MartinquaXD) [inline crates/solvers/src/infra/config/dex/oneinch/file.rs:27]: The `Liquidity` enum is a nice idea but according to the serde [docs](https://serde.rs/container-attrs.html#deny_unknown_fields) `deny_unknown_fields` doesn't play nice with `flatten`.

**Comment by nlordell** [comment]
We do have tests. For example, here are the 0x solver tests: https://github.com/cowprotocol/services/blob/main/crates/solvers/src/tests/zeroex/mod.rs

**Comment by cowfee** [comment]
Writing the tests.

**Comment by cowfee** [comment]
Added tests and fixed a bug that was previously present with expecting `gas` to be a string.

https://github.com/cowprotocol/services/pull/1518/commits/efc52cbbd1375dc908b965945559c0db26cbcbff

**Review by MartinquaXD — APPROVED** [review]

**Review by vkgnosis — APPROVED** [review]

**Review by nlordell — APPROVED** [review]

**Pull request merged**